src/langchain/llm-chat.ts

```diff
@@ -3,117 +3,186 @@ import {
   BaseChatModelParams,
 } from '@langchain/core/language_models/chat_models';
 import {
+  AIMessage,
+  AIMessageChunk,
   BaseMessage,
-  MessageType,
-  SystemMessage,
 } from '@langchain/core/messages';
 import { CallbackManagerForLLMRun } from '@langchain/core/callbacks/manager';
-import { ChatResult } from '@langchain/core/outputs';
+import { ChatGenerationChunk, ChatResult } from '@langchain/core/outputs';
+import { BaseLanguageModelCallOptions } from '@langchain/core/language_models/base';
+import merge from 'lodash/merge.js';
 
-import { InvalidInputError } from '../errors.js';
-import { concatUnique } from '../helpers/common.js';
-import type { RequiredPartial } from '../helpers/types.js';
-import { TextGenerationCreateOutput } from '../schema.js';
+import { Client, Configuration } from '../client.js';
+import {
+  TextChatCreateInput,
+  TextChatCreateStreamInput,
+  TextGenerationCreateInput,
+} from '../schema.js';
+import { InternalError, InvalidInputError } from '../errors.js';
 
-import { GenAIModel, GenAIModelOptions } from './llm.js';
+export type GenAIChatModelParams = BaseChatModelParams &
+  Required<
+    Pick<TextGenerationCreateInput & TextChatCreateStreamInput, 'model_id'>
+  > &
+  Pick<
+    TextGenerationCreateInput & TextChatCreateStreamInput,
+    'prompt_id' | 'parameters'
+  > & { model_id: string; configuration?: Configuration };
+export type GenAIChatModelOptions = BaseLanguageModelCallOptions &
+  Pick<GenAIChatModelParams, 'parameters'>;
 
-export type RolesMapping = RequiredPartial<
-  Record<
-    MessageType,
-    {
-      stopSequence: string;
-    }
-  >,
-  'system'
->;
+export class GenAIChatModel extends BaseChatModel<GenAIChatModelOptions> {
+  protected readonly client: Client;
 
-type Options = BaseChatModelParams &
-  GenAIModelOptions & {
-    rolesMapping: RolesMapping;
-  };
+  public readonly modelId: GenAIChatModelParams['model_id'];
+  public readonly promptId: GenAIChatModelParams['prompt_id'];
+  protected parameters: GenAIChatModelParams['parameters'];
 
-export class GenAIChatModel extends BaseChatModel {
-  readonly #model: GenAIModel;
-  readonly #rolesMapping: RolesMapping;
+  protected conversation: string | null = null;
```

Comment:
What's `conversation`? Is it like the ID of the conversation?
I would make the class stateless as it's done in other LLM implementations.